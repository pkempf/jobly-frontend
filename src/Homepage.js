import React, { useContext } from "react";
import Jumbotron from "react-bootstrap/Jumbotron";
import { Link } from "react-router-dom";
import UserContext from "./UserContext";

const Homepage = () => {
  const user = useContext(UserContext);
  return (
    <Jumbotron className="mt-3 text-center">
      <h1>
        <b>Jobly</b>
      </h1>
      <h4 className="mb-4">Let's get to work.</h4>
      {user.username ? (
        <h3>Welcome back, {user.username}!</h3>
      ) : (
        <span>
          <Link className="btn btn-primary mr-1" to="/login">
            Log in
          </Link>
          <Link className="btn btn-primary ml-1" to="/signup">
            Sign up
          </Link>
        </span>
      )}
    </Jumbotron>
  );
};

export default Homepage;
