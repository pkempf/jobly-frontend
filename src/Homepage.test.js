import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Homepage from "./Homepage";
import UserContext from "./UserContext";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{}}>
        <Homepage />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{}}>
        <Homepage />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
