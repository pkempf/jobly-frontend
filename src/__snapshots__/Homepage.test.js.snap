// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-3 text-center jumbotron"
  >
    <h1
      style="font-size: 72px;"
    >
      <strong>
        Jobly
      </strong>
    </h1>
    <h4
      class="my-4"
    >
      Let's get to work.
    </h4>
    <span>
      <a
        class="btn btn-primary mr-1"
        href="/login"
      >
        Log in
      </a>
      <a
        class="btn btn-primary ml-1"
        href="/signup"
      >
        Sign up
      </a>
    </span>
  </div>
</DocumentFragment>
`;
